src/zope/testbrowser/browser.py

```diff
@@ -847,7 +847,9 @@ def displayValue(self, value):
 
         values = []
         for key, titles in self._getOptions():
-            if any(t in value for t in titles):
+            if any(v in t
+                   for t in titles
+                   for v in value):
```

Comment:
What happens if I assing a string to `displayValue`?  I don't think character-by-character iteration is right here.